src/components/WalletModal/Option.tsx

```diff
@@ -158,30 +158,32 @@ interface OptionProps {
 }
 export default function Option({ connection }: OptionProps) {
   const { activationState, tryActivation } = useActivationState()
-  const [WC2PromptOpen, setWC2PromptOpen] = useState(false)
+  const [WC1PromptOpen, setWC1PromptOpen] = useState(false)
   const [accountDrawerOpen, toggleAccountDrawerOpen] = useAccountDrawer()
   const activate = () => tryActivation(connection, toggleAccountDrawerOpen)
 
   useEffect(() => {
-    if (!accountDrawerOpen) setWC2PromptOpen(false)
+    if (!accountDrawerOpen) setWC1PromptOpen(false)
   }, [accountDrawerOpen])
 
   const isSomeOptionPending = activationState.status === ActivationStatus.PENDING
   const isCurrentOptionPending = isSomeOptionPending && activationState.connection.type === connection.type
   const isDarkMode = useIsDarkMode()
 
-  const handleClickConnectViaWCv2 = (e: MouseEvent<HTMLButtonElement>) => {
+  const handleClickConnectViaWCv1 = (e: MouseEvent<HTMLButtonElement>) => {
     e.stopPropagation()
-    tryActivation(walletConnectV2Connection, () => {
-      setWC2PromptOpen(false)
+    tryActivation(walletConnectV1Connection, () => {
+      setWC1PromptOpen(false)
       toggleAccountDrawerOpen()
     })
   }
-  const handleClickOpenWCv2Tooltip = (e: MouseEvent<HTMLButtonElement>) => {
+  const handleClickOpenWCv1Tooltip = (e: MouseEvent<HTMLButtonElement>) => {
     e.stopPropagation()
-    setWC2PromptOpen(true)
+    setWC1PromptOpen(true)
   }
 
+  const showExtraMenuToggle = connection.type === ConnectionType.WALLET_CONNECT_V2 && !isCurrentOptionPending
```

Comment:
@JFrankfurt was this added as an extra check to fix a visual bug in the existing logic? lgtm, just wondering why it's the only addition here.

Reply:
This is already answered in the PR description, thank you!